### PR TITLE
Create / Update

### PR DESCRIPTION
**Pull request by dimitre**
This PR makes recursive file update outside the create / update block,
fix issue with projectName being treated differently for paths with trailing slashes or without
Adds new archs for RPI 4

### CODE REVIEW

**Comment by ofTheo** [comment]
So this should handle a project that's project folder doesn't exist as well as one where it has a project folder with src/ folder just as well? 

Looks like an error in macOS ci might need a PR to OF/OF to be merged first?

**Comment by dimitre** [comment]
Yes. I just not sure by a folder that doesn't exist. never tested this before
you mean PG should create a project folder if it doesn't exist?
about the test I think the cache is making it fail. will try to erase cache manually so it has to clone latest OF

**Comment by dimitre** [comment]
@ofTheo I've pushed two important changes here. 
One, PG had an error using make because OF_ROOT was being written in config.make with double quotes., fixed
other one, it was altering OF_ROOT only if it came from template, commented out ```str.rfind("# OF_ROOT =", 0) == 0```
and I've completed with checking also uncommented also, so you get the right OF_ROOT if you once decide to move the project somewhere else and run PG again.

File: commandLine/src/main.cpp [hunk 1]
```diff
@@ -244,7 +246,7 @@ void recursiveUpdate(const fs::path & path, ofTargetPlatform target) {
 	// second check if this is a folder that has src in it
 	if (isGoodProjectPath(path)) {
 		nProjectsUpdated++;
-		auto project = getTargetProject(target);
+//		auto project = getTargetProject(target);
```
Comment (ofTheo) [inline commandLine/src/main.cpp:249]: just curious why this was commented out?

Comment (dimitre) [reply commandLine/src/main.cpp:249]: updateProject function has the same code inside. so it was just hanging there

Comment (dimitre) [reply commandLine/src/main.cpp:249]: I should enable unused vars for this project

Comment (ofTheo) [reply commandLine/src/main.cpp:249]: ahh yes - expanding now I see it. Thanks!!

**Comment by ofTheo** [comment]
Awesome! Can you pull in PG/Master ? 
I added PG generation of examples folder to the build script - so we can make sure these changes can still generate examples without issue.

**Comment by dimitre** [comment]
Sure! let me know when changes are there and I pull in this one. no commits behind. maybe it is already up to date

**Pull request merged**